### ISSUE TITLE
Keras examples not training

### ISSUE THREAD

**Issue by NeonAutomata** [keras-team/keras#12412]
Hello, I am trying to setup Keras on Ubuntu 18.10 with Tensorflow CPU-training. I have installed Keras and Tensorflow with pip3 and copied and pasted the MLP for Binary Classification example from the [Keras docs](https://keras.io/getting-started/sequential-model-guide/#examples). However when I try to run this example I get the following output:

```
username@Computer:~/Stock$ python3 Test.py 
Using TensorFlow backend.
WARNING:tensorflow:From /home/username/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/username/.local/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:3680: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
WARNING:tensorflow:From /home/username/.local/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
Epoch 1/20
2019-03-07 12:43:21.899003: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2019-03-07 12:43:21.906434: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2591995000 Hz
2019-03-07 12:43:21.906786: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x26df370 executing computations on platform Host. Devices:
2019-03-07 12:43:21.906838: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
1000/1000 [==============================] - 0s 339us/step - loss: 2.4392 - acc: 0.0880
Epoch 2/20
1000/1000 [==============================] - 0s 16us/step - loss: 2.4063 - acc: 0.0890
Epoch 3/20
1000/1000 [==============================] - 0s 16us/step - loss: 2.3452 - acc: 0.1050
Epoch 4/20
1000/1000 [==============================] - 0s 24us/step - loss: 2.3304 - acc: 0.0970
Epoch 5/20
1000/1000 [==============================] - 0s 22us/step - loss: 2.3203 - acc: 0.1200
Epoch 6/20
1000/1000 [==============================] - 0s 20us/step - loss: 2.3172 - acc: 0.1010
Epoch 7/20
1000/1000 [==============================] - 0s 25us/step - loss: 2.3195 - acc: 0.1130
Epoch 8/20
1000/1000 [==============================] - 0s 20us/step - loss: 2.3126 - acc: 0.1110
Epoch 9/20
1000/1000 [==============================] - 0s 26us/step - loss: 2.3158 - acc: 0.0920
Epoch 10/20
1000/1000 [==============================] - 0s 26us/step - loss: 2.3042 - acc: 0.1130
Epoch 11/20
1000/1000 [==============================] - 0s 17us/step - loss: 2.3081 - acc: 0.1020
Epoch 12/20
1000/1000 [==============================] - 0s 17us/step - loss: 2.3064 - acc: 0.1130
Epoch 13/20
1000/1000 [==============================] - 0s 16us/step - loss: 2.3038 - acc: 0.1170
Epoch 14/20
1000/1000 [==============================] - 0s 15us/step - loss: 2.3097 - acc: 0.1070
Epoch 15/20
1000/1000 [==============================] - 0s 13us/step - loss: 2.3162 - acc: 0.0880
Epoch 16/20
1000/1000 [==============================] - 0s 24us/step - loss: 2.3101 - acc: 0.1040
Epoch 17/20
1000/1000 [==============================] - 0s 44us/step - loss: 2.3026 - acc: 0.1020
Epoch 18/20
1000/1000 [==============================] - 0s 41us/step - loss: 2.3051 - acc: 0.0910
Epoch 19/20
1000/1000 [==============================] - 0s 41us/step - loss: 2.3015 - acc: 0.1060
Epoch 20/20
1000/1000 [==============================] - 0s 44us/step - loss: 2.3068 - acc: 0.1100
100/100 [==============================] - 0s 1ms/step

```

The loss and accuracy seem random and **the amount of time per epoch seems suspiciously low**. Given that there are no obvious error messages I think the issue is probably a configuration issue. I have tried searching around for similar issues but couldn't find anything related. 

Any help would be greatly appreciated.

**Comment by joosephook** [comment]
I looked at the example and noticed this:
```python
# Generate dummy data
x_train = np.random.random((1000, 20))
y_train = np.random.randint(2, size=(1000, 1))
x_test = np.random.random((100, 20))
y_test = np.random.randint(2, size=(100, 1))
```
As you can see, the model is being trained with random data - there is no relationship between x and y.  Here are some [datasets](https://keras.io/datasets/).

**Comment by NeonAutomata** [comment]
Yikes, can't believe I missed that section. I've run some tests with the datasets you linked and they're looking much better. Cheers.
